### PR TITLE
Patch over some TensorFlow Python API changes.

### LINKED ISSUE #14695: UnicodeDecodeError: 'utf-8' codec can't decode byte 0xef in position 2: invalid continuation byte
### What happened?

I was trying the sample `samples/colab/edge_detection.ipynb`  but got this error:
`samples/colab/edge_detection.ipynb`
I'm not downloading  iree-compiler, iree-runtime,iree-tools-tf at tag "candidate-20230426.501",instead I build the iree from source

### What component(s) does this issue relate to?

_No response_

### Version information

iree：commit 562f2ed9a3a9d12de91792dff40d64f7a180ce95 (HEAD -> main, tag: candidate-20230816.17, origin/main, origin/latest-snapshot, origin/HEAD

tf:  tf-nightly                   2.14.0.dev20230705

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ScottTodd**
Working towards getting our sample Colab notebooks working again (https://github.com/openxla/iree/actions/workflows/samples.yml). I could try to update the Colab notebooks in-place, but it might just be easier to take an incremental approach - land these fixes that help with local builds, wait for a nightly release, debug the notebooks with that release build.

We're also short on presubmit / unit test coverage for the Python tools.

Resolves this discussion: https://github.com/openxla/iree/pull/14303#discussion_r1260364221
Fixes https://github.com/openxla/iree/issues/14695

### CODE REVIEW

File: compiler/bindings/python/iree/compiler/tools/tf.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def compile_saved_model(saved_model_dir: str, **kwargs):
         if options.import_only:
             if options.output_file:
                 return None
-            with open(tf_iree_input, "r") as f:
+            with open(tf_iree_input, "rb") as f:
```
Comment (ScottTodd) [inline compiler/bindings/python/iree/compiler/tools/tf.py:149]: Colab notebooks should also be updated to handle bytecode, such as here: https://github.com/openxla/iree/blob/fb38d2092cefa8aca53b1ed67c35829cb039b06b/samples/variables_and_state/variables_and_state.ipynb#L185-L198


This code will read a bytecode file and print it as text:

```python
from iree.compiler import ir

module = ir.Module.parse(module_mlir_bytecode, ir.Context())
module.dump()
```

**Comment by ScottTodd** [comment]
ping @jpienaar (welcome back :D)

File: compiler/bindings/python/iree/compiler/tools/tf.py (hunk 1 above)
Comment (ScottTodd) [reply compiler/bindings/python/iree/compiler/tools/tf.py:149]: https://github.com/openxla/iree/pull/14743 also added a command to `ir_tool` that can do this.

**Comment by ScottTodd** [comment]
Synced. Ping again, @jpienaar ?

**Review by jpienaar — APPROVED** [review]
Thanks for the ping

File: compiler/bindings/python/iree/compiler/tools/tf.py (hunk 1 above)
Comment (ScottTodd) [reply compiler/bindings/python/iree/compiler/tools/tf.py:149]: Updating notebooks in https://github.com/openxla/iree/pull/14901

**Pull request merged**